lib/stdlib/src/sets.erl

```diff
@@ -74,56 +84,74 @@
 
 -type set() :: set(_).
 
--opaque set(Element) :: #set{segs :: segs(Element)}.
+-opaque set(Element) :: #set{segs :: segs(Element)} | #{Element => ?VALUE}.
 
 %%------------------------------------------------------------------------------
 
 %% new() -> Set
 -spec new() -> set().
-new() ->
+new() -> #{}.
```

Comment:
After discussing this internally we'd like the old implementation to be the default in order not to break distribution: an application on OTP 24 may send a set to another one on OTP 23, which will crash if you send a map.

We'll change the default in OTP 26 when all supported versions support maps.

Reply:
Thanks! Any suggestions on how to call the `new/0` and `from_list/1` variants that return `maps`? Or we don't want to introduce such functions as they will be removed on OTP 26 anyway?

Reply:
Not at the moment, we'll take it up at the next technical board meeting.

Reply:
Sounds good, I will wait until a decision on this front then. In the worst case scenario, we keep them as `map_new/0` and `map_from_list/1` so we can use it within OTP and in tests, but we don't make them public.

Reply:
We've decided to add `new/1` and `from_list/2` that take an option list, where one can specify `{version,1}` for the old and `{version,2}` for the new version.

We'll change the default in OTP 26.